### ISSUE TITLE
GOIV unable to scan on Samsung S8+

### ISSUE THREAD

**Issue by Rebelpilot** [farkam135/GoIV#679]
So it seems that the weird proportions of the Samsung S8+ make it impossible for goiv to scan the pokemon in the game. I think it's due to the fact the game loads with a black bar on the top of the screen.

How can I help you guys look into this so it can be fixed? Thanks.

Here is a screencap: https://goo.gl/photos/47dTgpLLyHbjWRvV9

**Comment by nahojjjen** [comment]
I'm looking at issue #677 which seems similar; newer android phones seem to have weird screen ratios. I wonder if we could make a more general solution instead of creating a manual fix for each phone. Maybe a "setup" screen in settings where the user takes a sceenshot from pokemon go, and then manually marks the scan areas for each scanfield?

So basically, 
user presses "setup scanfields" in settings
User is told to open screenshot of Pogo
user is told to draw a rectangle on the screenshot where the HP is
User is told to draw rectangle on the screenshot where cp is
.... rest of scanning rectangles
User presses "done"

Hopefully I'll have time to test making this later tomorrow...

**Comment by Rebelpilot** [comment]
When you get a beta code, I can run a test on my S8+ to help verify.  Just send me a link to the download when ready.

**Comment by Rhodan71** [comment]
I would like to beta test on S8 + as well ;)

**Comment by nahojjjen** [comment]
So me and tripsixes talked / looked through the code, and agreed that it'd be best if GoIV could fix the scanning automatically, without user input. From what we could tell, the code should have worked if the game filled the entire screen.

I don't have an S8 or equivalent, so I wont be able to test during development. Anyways, the idea we discussed was to check the borders of the screen, and ignore black spaces during screen parameter setup, however this is problematic because we setup the screen parameters before initiating the screen recorder.

**Comment by Rhodan71** [comment]
@nahojjjen 

It isnt only the black spaces I think

Because the game FILLS the whole screen, you can run it with Samsung Game Tools, which is included with every Samsung S8, in Full screen mode.

No black spaces at all then

In Full screen Mode Go IV reads at least the WP, but it doesnt recognize the Pokemon Name and the other values.

 I think the issue is more the unusual 18.5:9  display ratio

**Comment by Kantana28** [comment]
Count me in to test any betas, I have a LG G6 with 18:9 screen ratio.  LG provides a scalar app to force 16:9, 16.7:9 or 18:9.  I've tried all 3 of those and only 18:9 works with latest GoIV but the scanner fails to find names, hp etc

**Comment by Rhodan71** [comment]
Any news on a new beta? I can test on my S8+

**Comment by Rebelpilot** [comment]
@nahojjjen here is the screen caps from my S8+ as requested:

https://drive.google.com/open?id=0BxcMSqZN1eCzZ3VfTnBZUE03Vzg

 2 showing the game in Full Screen Mode, 2 in regular Mode and the resolution sizes of the phone and what they can be set too:

**Comment by nahojjjen** [comment]
The s8 does not seem to follow the same scaling in pogo after all...


![msrmsnts](https://cloud.githubusercontent.com/assets/8545451/25554412/2227ac86-2ccd-11e7-835b-9997b1bdab05.png)

The arc-radius seems to be 20.88% of the screen height, instead of the normal 22.85%
Im guessing pogo stops scaling its contents width based on the phone height after a cap, and instead starts just showing more content at the bottom
The ratio of pokemon-display and info-card is also skewed, with S8 having 35% of the screen showing the pokemon, while normally its 38%


Anyways, I'm trying to make a "dirty" patch for the S8, but first of all, I need to see if I can detect if the application is running on an S8.
Here's a test apk I built (link is temporary)
https://drive.google.com/file/d/0BxUoqs3paaF0aUtjZVlwa2NCZ2s/view?usp=sharing

It should show a message "this looks like a samsung s8" if it... thinks its on a samsung s8. I've also tried to manually adjust the level-slider by measuring the values in photoshop and hard-coding them, but all the other scan values are still checking the wrong area. Could someone with an S8 check if anything's working?
